handler/implementations/sysDevicesVirtual.go

```diff
@@ -91,7 +91,7 @@ func (h *SysDevicesVirtual) Lookup(
 		return info, nil
 	}
 
-	return n.Stat()
+	return h.Service.GetPassThroughHandler().Lookup(n, req)
```

Comment:
I understand the need to invoke the `passthrough` handler for nodes within the `/sys/devices/virtual` hierarchy, but why do you need to do it for the `/sys/devices/virtual` node itself, which ends up requiring complex logic to avoid nsenter accesses during container initialization? IOW, why can't you always return a dummy node for this specific node as we do (e.g.) in `sysKernel.go` or `procSys.go`?

Reply:
We probably don't need it for the `/sys/device/virtual` node itself, but I added the call to the passthrough handler here because it's this same handler that deals with the subdirs of `/sys/devices/virtual` (except for the subdirs that have specific handlers). 

Alternatively we could create a new handler for `/sys/device/virtual/*` and have that handler call into the passthrough handler, but not sure that's worth it.

Reply:
No need to create a new handler. My point is that we can save the complexity associated with the need to skip the `passthrough` handler for bind-mounted resources (e.g., `/sys/devices/virtual`) during container initialization. And we do that by always returning a dummy node for lookups() targeting the `/sys/devices/virtual` node, before **and** after containers are initialized.

See an equivalent example here for `/proc/sys`:

```golang
func (h *ProcSys) Lookup(
	n domain.IOnodeIface,
	req *domain.HandlerRequest) (os.FileInfo, error) {

	logrus.Debugf("Executing Lookup() for req-id: %#x, handler: %s, resource: %s",
		req.ID, h.Name, n.Name())

	relpath, err := filepath.Rel(h.Path, n.Path())
	if err != nil {
		return nil, err
	}

	var resource = relpath

	if v, ok := h.EmuResourceMap[resource]; ok {
		if resource == "." {
			resource = "kernel"
		}
		info := &domain.FileInfo{
			Fname:    resource,
			Fmode:    v.Mode,
			FmodTime: time.Now(),
		}

		if v.Kind == domain.DirEmuResource {
			info.FisDir = true
		}

		return info, nil
	}

	return h.Service.GetPassThroughHandler().Lookup(n, req)
}
```

Btw, now that you're here, please fix this line below (cosmetic issue) by replacing 'kernel' with 'sys':

```golang
		if resource == "." {
			resource = "kernel"
		}
```

I believe an implementation such as the one I'm suggesting above, plus the existing [logic](https://github.com/nestybox/sysbox-fs/blob/a2631f69c62722c67dfd3aa97a8412b5c4db6a8a/mount/service.go#L43) to prevent recurrent sysbox-fs lookups for sysbox-fs bind-mounts, would be sufficient to address the second set of issues we previously discussed (related to recurrent i/o operations or transient issues during container init), without the need to add the extra complexity you created below (i.e., `ContainerIsRegistered()` invocations).

Reply:
Correct; I've followed your advice and updated the changeset, please take a look again.